pypdf/_reader.py

```diff
@@ -479,218 +270,6 @@ def _get_page(self, page_number: int) -> PageObject:
-            for kid in tree[PA.KIDS]:  # type: ignore
-                self.get_fields(kid.get_object(), retval, fileobj)
-
-    def _write_field(self, fileobj: Any, field: Any, field_attributes: Any) -> None:
-        field_attributes_tuple = FA.attributes()
-        field_attributes_tuple = (
-            field_attributes_tuple + CheckboxRadioButtonAttributes.attributes()
-        )
-
-        for attr in field_attributes_tuple:
-            if attr in (
-                FA.Kids,
-                FA.AA,
-            ):
-                continue
-            attr_name = field_attributes[attr]
-            try:
-                if attr == FA.FT:
-                    # Make the field type value more clear
-                    types = {
-                        "/Btn": "Button",
-                        "/Tx": "Text",
-                        "/Ch": "Choice",
-                        "/Sig": "Signature",
-                    }
-                    if field[attr] in types:
-                        fileobj.write(f"{attr_name}: {types[field[attr]]}\n")
-                elif attr == FA.Parent:
-                    # Let's just write the name of the parent
-                    try:
-                        name = field[attr][FA.TM]
-                    except KeyError:
-                        name = field[attr][FA.T]
-                    fileobj.write(f"{attr_name}: {name}\n")
-                else:
-                    fileobj.write(f"{attr_name}: {field[attr]}\n")
-            except KeyError:
-                # Field attribute is N/A or unknown, so don't write anything
-                pass
-
-    def get_form_text_fields(self, full_qualified_name: bool = False) -> Dict[str, Any]:
-        """
-        Retrieve form fields from the document with textual data.
-
-        Args:
-            full_qualified_name: to get full name.
-
-        Returns:
-            A dictionary. The key is the name of the form field,
-            the value is the content of the field.
-
-            If the document contains multiple form fields with the same name, the
-            second and following will get the suffix .2, .3, ...
-        """
-
-        def indexed_key(k: str, fields: Dict[Any, Any]) -> str:
-            if k not in fields:
-                return k
-            else:
-                return (
-                    k
-                    + "."
-                    + str(sum([1 for kk in fields if kk.startswith(k + ".")]) + 2)
-                )
-
-        # Retrieve document form fields
-        formfields = self.get_fields()
-        if formfields is None:
-            return {}
-        ff = {}
-        for field, value in formfields.items():
-            if value.get("/FT") == "/Tx":
-                if full_qualified_name:
-                    ff[field] = value.get("/V")
-                else:
-                    ff[indexed_key(cast(str, value["/T"]), ff)] = value.get("/V")
-        return ff
-
     def get_pages_showing_field(
```

Comment:
We cannot use the common implementation here?

Reply:
fixed duplication